src/explorer/explorer.ts

```diff
@@ -52,7 +52,14 @@ export function registerExplorerOpen(): vscode.Disposable {
       }
 
       try {
-        await vscode.window.showTextDocument(uri, { preview: usePreview });
+        if (uri.scheme === OBJECTSCRIPT_FILE_SCHEMA) {
+          // This scheme is implemented by our DocumentContentProvider, which always returns text.
+          // If the server supplied binary data our provider substitutes a text explanation of how to work with binary content.
+          await vscode.window.showTextDocument(uri, { preview: usePreview });
+        } else {
+          //  This allows use of binary editors such as the Luna Paint extension
+          await vscode.commands.executeCommand("vscode.open", uri, { preview: usePreview });
```

Comment:
I'm a little concerned about this line. Does `vscode.open` throw if `uri` can't be opened? If not then my project removal code in the catch block won't work.

Reply:
I have pushed a change for this case.

Reply:
Thanks John

Reply:
Actually, that doesn't fix the issue because now binary files that do exist are treated as if they don't exist. I think what we need to do is revert your last commit and before we call `vscode.open`, call `await vscode.workspace.file.stat(uri)`, which will throw `FileNotFound` if it doesn't exist.